api/util/queryset.py

```diff
@@ -0,0 +1,17 @@
+from django.core.paginator import Paginator
+
+
+def iterator_with_prefetch(queryset, chunk_size=2000):
```

Comment:
not sure if we should add test for this?

Reply:
As discussed, let's add a test and make sure that we verify the number of queries that are made.

Reply:
I have added that test to make sure we actually do prefetch data(per page) and are not making extra queries to access pre-fetched objects.